### DISCUSSION TITLE
how config kura project in eclipse in order to  run or debug

### DISCUSSION THREAD

**Discussion by chsunnyboy** [eclipse-kura/kura#5066] [Q&A]
i do not know how to config the kura project in eclipse to make it run or debug, who knows that ?  appreciate your methods  very much

**Comment by mattdibi** [comment]
Hello there @chsunnyboy,

you might want to take a look at the Kura documentation page:
- [Running the Eclipse Kura Emulator](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/development-environment-setup/#run-the-eclipse-kura-emulator)
- [Remote debugging on target platform](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/remote-debugging-on-target-platform/)
- [Deploy and Debug Applications](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/deploy-and-debug-applications/)

Hope this helps.

**Comment by chsunnyboy** [comment]
> Hello there @chsunnyboy,
> 
> you might want to take a look at the Kura documentation page:
> 
> * [Running the Eclipse Kura Emulator](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/development-environment-setup/#run-the-eclipse-kura-emulator)
> * [Remote debugging on target platform](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/remote-debugging-on-target-platform/)
> * [Deploy and Debug Applications](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/deploy-and-debug-applications/)
> 
> Hope this helps.


thanks anyway, but I have tried the two emulator below,but in the runtime there were many bundles were not resolved，and there were still many errors about import-package in the MANIFEAST.MF file  like the imgs

Eclipse Kura Emulator    
Kura_Emulator

<img width="948" alt="企业微信截图_17017598571686" src="https://github.com/eclipse/kura/assets/24762732/808dcc66-738d-4661-9266-0bce017e8c34">


<img width="593" alt="企业微信截图_17017606061711110" src="https://github.com/eclipse/kura/assets/24762732/776a56e0-960c-49ea-9535-71ce0b03cdf1">

**Comment by mattdibi** [comment]
@salvatore-coppola any thought on this?

Meanwhile I'm moving this to Discussion which is a better fit

**Comment by salvatore-coppola** [comment]
> > Hello there @chsunnyboy,
> > you might want to take a look at the Kura documentation page:
> > 
> > * [Running the Eclipse Kura Emulator](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/development-environment-setup/#run-the-eclipse-kura-emulator)
> > * [Remote debugging on target platform](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/remote-debugging-on-target-platform/)
> > * [Deploy and Debug Applications](https://eclipse.github.io/kura/docs-release-5.4/java-application-development/deploy-and-debug-applications/)
> > 
> > Hope this helps.
> 
> thanks anyway, but I have tried the two emulator below,but in the runtime there were many bundles were not resolved，and there were still many errors about import-package in the MANIFEAST.MF file like the imgs
> 
> Eclipse Kura Emulator Kura_Emulator
> <img alt="企业微信截图_17017598571686" width="948" src="https://private-user-images.githubusercontent.com/24762732/287955250-808dcc66-738d-4661-9266-0bce017e8c34.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MTc3MjQsIm5iZiI6MTcwNjcxNzQyNCwicGF0aCI6Ii8yNDc2MjczMi8yODc5NTUyNTAtODA4ZGNjNjYtNzM4ZC00NjYxLTkyNjYtMGJjZTAxN2U4YzM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE2MTAyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlZWU2ZDc3ZWI3NWJkM2NiNmNlYTJlMWNkNzUyZWEwNWY1YzkxNGI1ZmY5YTU2NGI0YzczZTY0MmI2ZWMxMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MIwAWE3yihATjPRFx_8ahhKN_EW_qco5y3bw0l8OO6w"> <img alt="企业微信截图_17017606061711110" width="593" src="https://private-user-images.githubusercontent.com/24762732/287955199-776a56e0-960c-49ea-9535-71ce0b03cdf1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3MTc3MjQsIm5iZiI6MTcwNjcxNzQyNCwicGF0aCI6Ii8yNDc2MjczMi8yODc5NTUxOTktNzc2YTU2ZTAtOTYwYy00OWVhLTk1MzUtNzFjZTBiMDNjZGYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDE2MTAyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwNzE3NWZmZDg2OGZlODRjODI1ZGVkMThkYzY1MmY4ZTlhYWQ5NTZhNGRiNjRhYzVlODc3MDJlZjZiNzRkYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Wqmc_jiEHCxiUFIWHmCCyOLXO88k9aqa9-lGAh4KjsE">

Hi @chsunnyboy , sorry for the late reply.

I suppose you are using the [latest user workspace](https://www.eclipse.org/downloads/download.php?file=/kura/releases/5.4.0/user_workspace_archive_5.4.0.zip) right ?
